node/pkg/lifecycle/startup/startup.go

```diff
@@ -112,6 +131,23 @@ func Run() {
 		}
 	}
 
+	// Make sure that this host's BlockAffinity resources are upgraded to add
+	// labels for efficient lookup.  CNI plugin also does this on the first
+	// allocation after upgrade.  Doing it here too handles some corner cases,
+	// such as calico-node being downgraded, doing some allocations and then,
+	// upgrading again.
+	upgradeCtx, cancel := context.WithTimeout(ctx, 90*time.Second)
```

Comment:
How confident are we that 90s will be enough to do this for large clusters? Probably pretty confident, given we only need to list the affinities once and we both shouldn't hit many conflicts (just with CNI potentially) and shouldn't have to make too many Update() calls, since each node handles its own?

Reply:
Pretty confident:
* List should take 2-3s at most
* We have a default blocks-per-host limit of 20 so we're talking about 20 writes.
Order of magnitude should be same as time to do pre-fix IPAM :-)